src/EventSubscriber/EntityWorkspaceMigrateSubscriber.php

```diff
@@ -0,0 +1,144 @@
+ * will be deleted and this is possible to get duplicated entities.
+ */
+class EntityWorkspaceMigrateSubscriber implements EventSubscriberInterface {
+
+  /**
+   * An array of entity types enabled in Multiversion.
+   *
+   * @var array $enabledEntityTypes
+   */
+  protected $enabledEntityTypes;
+
+  /**
+   * Default workspace ID.
+   *
+   * @var int $defaultWorkspaceId
+   */
+  protected $defaultWorkspaceId;
+
+  /**
+   * Active workspace.
+   *
+   * @var \Drupal\multiversion\Entity\Workspace $activeWorkspace
+   */
+  protected $activeWorkspace;
+
+  /**
+   * The workspace manager.
+   *
+   * @var \Drupal\multiversion\Workspace\WorkspaceManagerInterface $workspaceManager
+   */
+  protected $workspaceManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity query.
+   *
+   * @var \Drupal\Core\Entity\Query\QueryFactory $entityQuery
+   */
+  protected $entityQuery;
+
+  /**
+   * The container.
+   *
+   * @var \Symfony\Component\DependencyInjection\ContainerInterface $container
+   */
+  protected $container;
+
+  /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   */
+  protected $configFactory;
+
+  /**
+   * EntityWorkspaceMigrateSubscriber constructor.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Tne entity type manager.
+   * @param \Drupal\multiversion\Workspace\WorkspaceManagerInterface $workspace_manager
+   *   The workspace manager.
+   * @param \Drupal\Core\Entity\Query\QueryFactory $entity_query
+   *   The query factory.
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The container.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, WorkspaceManagerInterface $workspace_manager, QueryFactory $entity_query, ContainerInterface $container, ConfigFactoryInterface $config_factory) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->workspaceManager = $workspace_manager;
+    $this->entityQuery = $entity_query;
+    $this->container = $container;
+    $this->configFactory = $config_factory;
+    $this->enabledEntityTypes = $this->configFactory->get('multiversion.settings')->get('enabled_entity_types');
+    $this->defaultWorkspaceId = $this->container->getParameter('workspace.default');
+    $this->activeWorkspace = $this->workspaceManager->getActiveWorkspace();
+  }
+
+  /**
+   * Delete all entities that do not belong to the default workspace.
+   */
+  public function onPreMigrate() {
+    $workspaces = $this->workspaceManager->loadMultiple();
```

Comment:
Will it throw a notice when workspace module is not enabled?

Reply:
It won't since the workspace manager is a part of the multiversion module.